infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/StateTransferManager.java

```diff
@@ -106,6 +107,47 @@ TransferSegmentRangeSingle getUnknownAddressesInRangeForRange(TransferSegmentRan
                 .build();
     }
 
+    static boolean sequential(long first, long second) {
+        return first + 1 == second;
+    }
+
+    static Stream<List<Long>> splitNonSequentialAddresses(List<Long> batch) {
+        if (batch.size() < 2) {
+            return Stream.of(batch);
+        }
+        List<List<Long>> lists = new ArrayList<>();
+        List<Long> sequentialList = new ArrayList<>();
+        sequentialList.add(batch.get(0));
```

Comment:
please add checks that batch is not empty

Reply:
On the first line, if the size is less than 2 it returns.